packages/mongodb-memory-server-core/src/util/MongoBinaryDownload.ts

```diff
@@ -214,6 +217,74 @@ export default class MongoBinaryDownload {
     return extractDir;
   }
 
+  /**
+   * Extract a .tar.gz archive
+   * @param mongoDBArchive Archive location
+   * @param extractDir Directory to extract to
+   * @param filter Method to determine which files to extract
+   */
+  async extractTarGz(
+    mongoDBArchive: string,
+    extractDir: string,
+    filter: (file: string) => boolean
+  ): Promise<void> {
+    const extract = tar.extract();
+    extract.on('entry', (header, stream, next) => {
+      if (filter(header.name)) {
+        stream.pipe(
+          fs.createWriteStream(path.resolve(extractDir, path.basename(header.name)), {
+            mode: 0o775,
+          })
+        );
+      }
+
+      stream.on('end', () => next());
+      stream.resume();
+    });
+
+    return new Promise((resolve, reject) => {
+      extract.on('finish', () => resolve());
+      extract.on('error', (e) => reject(e));
+      fs.createReadStream(mongoDBArchive)
+        .pipe(createUnzip())
+        .pipe(extract);
```

Comment:
A suggestion for [@nodkz comment](https://github.com/nodkz/mongodb-memory-server/pull/274/files/61932b1fef3c0c7e6d64981ca73dc2fa49a01a6f#r389244980)
(couldnt add this to his, because this needs to select multiple lines)
```suggestion
      fs.createReadStream(mongoDBArchive)
        .on('error', reject)
        .pipe(createUnzip())
        .on('error', reject)
        .pipe(extract);
        .on('error', reject)
        .on('finish', resolve)
```